sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_path_client.py

```diff
@@ -392,6 +392,156 @@ def get_access_control(self, upn=None,  # type: Optional[bool]
+            The default when unspecified is 2000.
+        :keyword int timeout:
+            The timeout parameter is expressed in seconds.
+        :return: A summary of the recursive operations, including the count of successes and failures,
+            as well as a continuation token in case the operation was terminated prematurely.
+        :rtype: :class:`~azure.storage.filedatalake.models.AccessControlChanges`
+        """
+        if not acl:
+            raise ValueError("The Access Control List must be set for this operation")
+
+        options = self._set_access_control_recursive_options('set', acl=acl, **kwargs)
+        return self._set_access_control_internal(options, progress_callback)
+
+    def update_access_control_recursive(self,
+                                        acl=None,  # type: Optional[str]
+                                        progress_callback=None,
+                                        **kwargs):
+        # type: (...) -> Dict[str, Union[str, datetime]]
+        """
+        Modifies the Access Control on a path and sub-paths.
+
+        :param acl:
+            Modifies POSIX access control rights on files and directories.
+            The value is a comma-separated list of access control entries. Each
+            access control entry (ACE) consists of a scope, a type, a user or
+            group identifier, and permissions in the format
+            "[scope:][type]:[id]:[permissions]".
+        :type acl: str
+        :param progress_callback:
+            Callback where the caller can track progress of the operation
+            as well as collect paths that failed to change Access Control.
+        :type progress_callback: func(~azure.storage.filedatalake._generated.models.SetAccessControlRecursiveResponse)
+        :keyword str continuation:
+            Optional continuation token that can be used to resume previously stopped operation.
+        :keyword int batch_size:
+            Optional. If data set size exceeds batch size then operation will be split into multiple
+            requests so that progress can be tracked. Batch size should be between 1 and 2000.
+            The default when unspecified is 2000.
+        :keyword int timeout:
+            The timeout parameter is expressed in seconds.
+        :return: A summary of the recursive operations, including the count of successes and failures,
+            as well as a continuation token in case the operation was terminated prematurely.
+        :rtype: :class:`~azure.storage.filedatalake.models.AccessControlChanges`
+        """
+        if not acl:
+            raise ValueError("The Access Control List must be set for this operation")
+
+        options = self._set_access_control_recursive_options('modify', acl=acl, **kwargs)
+        return self._set_access_control_internal(options, progress_callback)
+
+    def remove_access_control_recursive(self,
```

Comment:
Why do we call this `remove` when it's still a set operation? That seems confusing.

Reply:
The ACL in a remove operation has a different format, it's still calling the same underlying API, just with mode='remove'.

Reply:
Do you see another option to make this clearer? I agree that it can be confusing.

Reply:
hmmm maybe it's not the name of the function - maybe we can change the name of the parameter to achieve more clarity? If the parameter name is "access_control_entries" - it defines it as a list of entries that we're removing the permissions from.
We could even extend that concept across all three set/update/remove function:
`set_access_control_recursive(entries: List[str], permissions: List[str])`
`update_access_control_recursive(entries: List[str], permissions: List[str])`
`remove_access_control_recursive(entries: List[str])`
Thoughts?

Reply:
Sorry I didn't understand, how do we separate the entries exactly?

Here is a description of the header:

> Optional and valid only for the setAccessControl operation. Sets POSIX access control rights on files and directories. The value is a comma-separated list of access control entries that fully replaces the existing access control list (ACL). Each access control entry (ACE) consists of a scope, a type, a user or group identifier, and permissions in the format "[scope:][type]:[id]:[permissions]". The scope must be "default" to indicate the ACE belongs to the default ACL for a directory; otherwise scope is implicit and the ACE belongs to the access ACL. There are four ACE types: "user" grants rights to the owner or a named user, "group" grants rights to the owning group or a named group, "mask" restricts rights granted to named users and the members of groups, and "other" grants rights to all users not found in any of the other entries. The user or group identifier is omitted for entries of type "mask" and "other". The user or group identifier is also omitted for the owner and owning group. The permission field is a 3-character sequence where the first character is 'r' to grant read access, the second character is 'w' to grant write access, and the third character is 'x' to grant execute permission. If access is not granted, the '-' character is used to denote that the permission is denied. For example, the following ACL grants read, write, and execute rights to the file owner and john.doe@contoso, the read right to the owning group, and nothing to everyone else: "user::rwx,user:john.doe@contoso:rwx,group::r--,other::---,mask=rwx". Invalid in conjunction with x-ms-permissions.

Reply:
Yeah - you're right - that's unpleasant.... hmm thanks for the docs, I will keep thinking :)

Reply:
So I've been thinking about this a bit further and I'm curious about where we expect these strings to come from.
Are they likely to be found - pre-formatted like this - from some file system API? Or will users be building these on their own?
Do you think there would be any benefit to supporting taking a list of strings here so users don't have to format it?
```
def set_access_control_recursive(paths, **kwargs):
    ...

directory_client.set_access_control_recursive(['user::rwx', 'group::r-x', 'other::rwx'])
```

Reply:
Another question - what happens if I were to supply this string `"user::rwx,group::r-x,other::rwx"` to the `remove` API? Would it fail? Or would the permissions values just be ignored?
If it fails - what kind of error do you get?